File: app/src/main/java/com/notpr/emberlist/parsing/BulkQuickAdd.kt
package com.notpr.emberlist.parsing

private val commonListMarkerRegex = Regex(
    "^\\s*(?:(?:[-*•◦▪‣]\\s*)?(?:\\[(?: |x|X)\\]|☐|☑|✅)\\s+|(?:[-*•◦▪‣])\\s+)"
)

fun extractBulkQuickAddLines(input: String): List<String> {
    return input
        .lineSequence()
        .map { it.trim() }
        .filter { it.isNotBlank() }
        .map(::stripCommonListMarker)
        .filter { it.isNotBlank() }
        .toList()
}

fun stripCommonListMarker(line: String): String {
    return line.replace(commonListMarkerRegex, "").trim()
}

fun shouldPromptBulkQuickAdd(input: String): Boolean {
    return extractBulkQuickAddLines(input).size > 1
}

File: web/src/lib/bulkQuickAdd.ts
const COMMON_LIST_MARKER_REGEX =
  /^\s*(?:(?:[-*•◦▪‣]\s*)?(?:\[(?: |x|X)\]|☐|☑|✅)\s+|(?:[-*•◦▪‣])\s+)/;

export function stripCommonListMarker(line: string): string {
  return line.replace(COMMON_LIST_MARKER_REGEX, '').trim();
}

export function extractBulkQuickAddLines(input: string): string[] {
  return input
    .split(/\r?\n/)
    .map(line => line.trim())
    .filter(Boolean)
    .map(stripCommonListMarker)
    .filter(Boolean);
}

export function shouldPromptBulkQuickAdd(input: string): boolean {
  return extractBulkQuickAddLines(input).length > 1;
}
